doc/source/notebooks/pandas.ipynb

```diff
@@ -1,2066 +1,2071 @@
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>name</th>\n",
+       "      <th>age</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>2bd806c97f0e00af1a1fc3328fa763a9269723c8db8fac...</td>\n",
+       "      <td>18</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>81b637d8fcd2c6da6359e6963113a1170de795e4b725b8...</td>\n",
+       "      <td>19</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>4c26d9074c27d89ede59270c0ac14b71e071b15239519f...</td>\n",
+       "      <td>19</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                                                name  age\n",
+       "0  2bd806c97f0e00af1a1fc3328fa763a9269723c8db8fac...   18\n",
+       "1  81b637d8fcd2c6da6359e6963113a1170de795e4b725b8...   19\n",
+       "2  4c26d9074c27d89ede59270c0ac14b71e071b15239519f...   19"
+      ]
+     },
+     "execution_count": 26,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pd_df.apply(\n",
+    "    lambda df: asdict(hide_name(df[\"name\"], df[\"age\"])),\n",
+    "    axis=1, \n",
+    "    result_type=\"expand\"\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "But there are still some important **differences** between the two cases:\n",
+    "\n",
+    "- In pandas, what we apply to a `DataFrame` is a Python function, while in GreenplumPython, a Python function must be converted to a database function before being applied to a `DataFrame`.\n",
+    "- pandas supports applying a function along different axes, while GreenplumPython only supports applying a function to each row due to the limitation of the database system behind.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Data Grouping\n",
+    "\n",
+    "Like ordering, Grouping data based on distinct set of values of columns can also facilitate analytical tasks.\n",
+    "\n",
+    "Data grouping is often associated with aggregate functions to obtain data summaries. \n",
+    "\n",
+    "For example, suppose we want to the number of students of different age. In pandas, we can do"
```

Comment:
For example, suppose we want to get the students' count of different ages. In pandas, we can do: